mgmt/rest/v2/plugin.go

```diff
@@ -244,13 +280,40 @@ func (s *apiV2) loadPlugin(w http.ResponseWriter, r *http.Request, _ httprouter.
 			default:
 				ec = 500
 			}
+			cleanUpTempFiles(rp)
 			Write(ec, rb, w)
 			return
 		}
+		cleanUpTempFiles(rp)
 		Write(201, catalogedPluginBody(r.Host, pl), w)
 	}
 }
 
+func isTLSEnabled(cert, key string) bool {
+	if cert != "" && key != "" {
+		return true
+	}
+	return false
+}
+
+func createTempFile(b []byte, fn string) (string, error) {
+	file, err := ioutil.TempFile(os.TempDir(), fn)
+	if err != nil {
+		return "", err
+	}
+	_, err = file.Write(b)
+	if err != nil {
+		return "", err
+	}
+	return file.Name(), nil
+}
+
+func cleanUpTempFiles(rp *core.RequestedPlugin) {
+	defer os.Remove(rp.CACertPaths())
+	defer os.Remove(rp.CertPath())
+	defer os.Remove(rp.KeyPath())
```

Comment:
In my opinion using "defer" here does not have sense - @candysmurf, maybe it is something which is no longer valid and you missed removing that. Could you look on that again?